cookbooks/cdo-repository/libraries/git.rb

```diff
@@ -0,0 +1,62 @@
+# Patch Chef::Provider::Git to support creating/updating shallow clones.
+class Chef
+  class Provider
+    class Git < Chef::Provider
+
+      def clone
```

Comment:
Not absolutely critical to patch, I just found it quicker to modify the existing Chef resource in this way rather than figure out how to build up a new [custom Chef resource](https://docs.chef.io/custom_resources.html) that reproduces the original behavior and includes the few patches here.

I can figure out how to refactor this into a separate resource extending the original (and it would be helpful to know how to create custom resources anyway).